src/NuGet.Clients/NuGet.Console/ScriptExecutor.cs

```diff
@@ -178,7 +182,11 @@ private async Task ExecuteScriptCoreAsync(ScriptExecutionRequest request)
             // to switch to powershell pipeline execution thread. In order not to block the UI thread,
             // go off the UI thread. This is important, since, switches to UI thread,
             // using SwitchToMainThreadAsync will deadlock otherwise
-            await Task.Run(() => host.Execute(console, request.BuildCommand(), request.BuildInput()));
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            _pumpingJTF.Run(async () =>
+            {
+                await Task.Run(() => host.Execute(console, request.BuildCommand(), request.BuildInput()));
+            });
```

Comment:
To me, it seems this:

1. It switches to main thread
2. Uses pumping JTF to run synchronous code, because, sometime, PowerShell `install.ps1` scripts access DTE object, which it need access to VS UI thread.
3. `await Task.Run( ... )` creates another thread to run PowerShell code, which, according to comment above, blocks its running thread.

Would it be better to use the following?

```suggestion
            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
            _pumpingJTF.Run(async () =>
            {
                await TaskScheduler.Default;
                host.Execute(console, request.BuildCommand(), request.BuildInput());
            });
```

I am suggesting using `TaskScheduler.Default` because I read somewhere `Task.Run(...)` sometimes leads to deadlocks.

Could you please update comment above to avoid confussion in 'going off the UI thread` and the line below that switches to UI/Main thread?